code/modules/overmap/_overmap.dm

```diff
@@ -0,0 +1,143 @@
+/datum/overmap
+	var/name
+	var/assigned_z
+
+	var/event_areas = 11
+	var/map_size_x = 20
+	var/map_size_y = 20
+
+	var/map_turf_type = /turf/space
+	var/map_area_type = /area/overmap
+	var/list/valid_event_types
+	var/empty_z_level_turf
+
+/datum/overmap/New(var/_name)
+
+	name = _name
+
+	if(!map_turf_type)
+		map_turf_type = world.turf
+
+	if(!name)
+		PRINT_STACK_TRACE("Unnamed overmap datum instantiated: [type]")
+
+	if(global.overmaps_by_name[name])
+		PRINT_STACK_TRACE("Duplicate overmap datum instantiated: [type], [name], [overmaps_by_name[name]]")
+	global.overmaps_by_name[name] = src
+
+	generate_overmap()
+
+	if(!assigned_z)
+		PRINT_STACK_TRACE("Overmap datum generated null assigned z_level.")
+
+	if(global.overmaps_by_z["[assigned_z]"])
+		PRINT_STACK_TRACE("Duplicate overmap datum instantiated for z-level: [type], [assigned_z], [overmaps_by_name[name]]")
+	global.overmaps_by_z["[assigned_z]"] = src
+
+	for(var/event_type in subtypesof(/datum/overmap_event))
+		var/datum/overmap_event/event = event_type
+		if(initial(event.overmap_id) == name)
+			LAZYADD(valid_event_types, event_type)
+
+	..()
+
+/datum/overmap/proc/generate_overmap()
+	testing("Building overmap [name]...")
+	INCREMENT_WORLD_Z_SIZE
+	assigned_z = world.maxz
+	testing("Putting [name] on [assigned_z].")
+
+	var/area/overmap/A = new map_area_type
+	A.name = name
```

Comment:
need to add it to area purity test exemptions

Reply:
Why was this not failing before?

Reply:
Because it wasn't a thing before, it's new code? Overmap area just used default name

Reply:
Removed the name setting, overmap datums can just define their own areas properly.